src/Driver/PDO/Connection.php

```diff
@@ -88,13 +90,45 @@ public function quote(string $value): string
         return $this->connection->quote($value);
     }
 
-    public function lastInsertId(): string
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
     {
+        $driverName = $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME);
+
+        if ($driverName === 'oci') {
```

Comment:
Instead of presuming the driver's incapability of fulfilling the API, we should let the driver explicitly report that. See: https://github.com/doctrine/dbal/blob/f18aee0a2b209662370de14f3a5268553c9b239e/tests/Functional/WriteTest.php#L366-L369